Code/EnginePlugins/OpenXRPlugin/OpenXRSingleton.cpp

```diff
@@ -0,0 +1,1122 @@
+    m_pCameraToSynchronize->SetStereoProjection(projLeft, projRight, fAspectRatio);
+    m_uiSettingsModificationCounter = m_pCameraToSynchronize->GetSettingsModificationCounter();
+  }
+
+  // Update camera view
+  {
+    ezTransform add;
+    add.SetIdentity();
+    ezView* pView = nullptr;
+    if (ezRenderWorld::TryGetView(m_hView, pView))
+    {
+      if (const ezWorld* pWorld = pView->GetWorld())
+      {
+        EZ_LOCK(pWorld->GetReadMarker());
+        if (const ezStageSpaceComponentManager* pStageMan = pWorld->GetComponentManager<ezStageSpaceComponentManager>())
+        {
+          if (const ezStageSpaceComponent* pStage = pStageMan->GetSingletonComponent())
+          {
+            ezEnum<ezXRStageSpace> stageSpace = pStage->GetStageSpace();
+            if (m_StageSpace != stageSpace)
+              SetStageSpace(pStage->GetStageSpace());
+            add = pStage->GetOwner()->GetGlobalTransform();
+          }
+        }
+      }
+    }
+
+    {
+      // Update device state (average of both eyes).
+      ezQuat rot;
+      rot.SetIdentity();
+      rot.SetSlerp(
+        ConvertOrientation(m_views[0].pose.orientation), ConvertOrientation(m_views[1].pose.orientation), 0.5f);
+      const ezVec3 pos =
+        ezMath::Lerp(ConvertPosition(m_views[0].pose.position), ConvertPosition(m_views[1].pose.position), 0.5f);
+
+      m_Input->m_DeviceState[0].m_vGripPosition = pos;
+      m_Input->m_DeviceState[0].m_qGripRotation = rot;
+      m_Input->m_DeviceState[0].m_Type = ezXRDeviceType::HMD;
+      m_Input->m_DeviceState[0].m_bGripPoseIsValid = true;
+      m_Input->m_DeviceState[0].m_bDeviceIsConnected = true;
+    }
+
+    // Set view matrix
+    {
+      const ezMat4 mStageTransform = add.GetAsMat4();
+      const ezMat4 poseLeft = mStageTransform * ConvertPoseToMatrix(m_views[0].pose);
+      const ezMat4 poseRight = mStageTransform * ConvertPoseToMatrix(m_views[1].pose);
+
+      // EZ Forward is +X, need to add this to align the forward projection
+      const ezMat4 viewMatrix =
+        ezGraphicsUtils::CreateLookAtViewMatrix(ezVec3::ZeroVector(), ezVec3(1, 0, 0), ezVec3(0, 0, 1));
+      const ezMat4 mViewTransformLeft = viewMatrix * poseLeft.GetInverse();
+      const ezMat4 mViewTransformRight = viewMatrix * poseRight.GetInverse();
+
+      m_pCameraToSynchronize->SetViewMatrix(mViewTransformLeft, ezCameraEye::Left);
+      m_pCameraToSynchronize->SetViewMatrix(mViewTransformRight, ezCameraEye::Right);
+    }
+  }
+
+  {
+    // put the camera orientation into the sound listener and enable the listener override mode
+    if (ezSoundInterface* pSoundInterface = ezSingletonRegistry::GetSingletonInstance<ezSoundInterface>())
+    {
+      pSoundInterface->SetListener(-1, m_pCameraToSynchronize->GetCenterPosition(),
+        m_pCameraToSynchronize->GetCenterDirForwards(), m_pCameraToSynchronize->GetCenterDirUp(), ezVec3::ZeroVector());
+    }
+  }
```

Comment:
Override mode is for the editor. It would be better to have a head-pose component that can be attached to the player prefab, because that's where you typically have the listener component.

Reply:
Ok works now without it thanks to the listener component.